test/parallel/test-http-client-timeout-option-listeners.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+require('../common');
+const http = require('http');
+const assert = require('assert');
+
+const agent = new http.Agent({ keepAlive: true });
+
+const server = http.createServer((req, res) => {
+  res.end('');
+});
+
+const options = {
+  agent,
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
+  path: '/',
+  timeout: 10
+};
+
+server.listen(0, options.host, (e) => {
+  options.port = server.address().port;
+
+  doRequest().then(doRequest)
+    .then((numListeners) => {
+      assert.strictEqual(numListeners, 1,
+                         'Should be a single timeout listener on the socket');
+      server.close();
+      agent.destroy();
+    })
+    .catch((e) => {
+      console.log(e);
```

Comment:
I could remove the whole catch clause and add a 
```
process.on('unhandledRejection', function () {
  common.fail('A promise rejection was unhandled')
});
```
Would that be ok?

I want to ensure that the test never hangs